rust/signed_doc/src/metadata/mod.rs

```diff
@@ -231,33 +232,77 @@ impl Display for Metadata {
     }
 }
 
-impl TryFrom<&Metadata> for coset::Header {
-    type Error = anyhow::Error;
+impl<C> minicbor::Encode<C> for Metadata {
+    fn encode<W: minicbor::encode::Write>(
+        &self, e: &mut minicbor::Encoder<W>, _ctx: &mut C,
+    ) -> Result<(), minicbor::encode::Error<W::Error>> {
+        let number_of_fields = [
+            self.0.content_type.is_some(),
+            self.0.content_encoding.is_some(),
+            self.0.doc_type.is_some(),
+            self.0.id.is_some(),
+            self.0.ver.is_some(),
+            self.extra().doc_ref().is_some(),
```

Comment:
It feels like an unnecessary complication to put these under `extra()` they are all protected headers in our specification, they are encoded in the same map, the only distinction is some are optional, and some aren't.  (and one is pre-defined by the COSE standard).